frontend/public/components/dashboard/dashboards-page/cluster-dashboard/cluster-dashboard.tsx

```diff
@@ -26,15 +29,24 @@ export const ClusterDashboard: React.FC<{}> = () => {
     InfrastructureModel,
     'cluster',
   );
-  // [TODO](sahil143): use sync capability here
+
+  const [consoleQuickStarts, consoleQuickStartsLoaded] = useK8sWatchResource<QuickStart[]>({
+    kind: referenceForModel(QuickStartModel),
+    isList: true,
+  });
```

Comment:
Since we are already loading quick starts here, I'm wondering if we can remove QuickStartsLoader from this file which does the same thing with additional permission checks. Can we create a hook that can fetch the quick starts and do permission checks as well?
Only checking for availability of quick starts seems flawed because there can be quickstarts that don't pass the permission check and if those are the only quickstarts available then the card might have zero quick starts and the gap will be still available.
cc @rohitkrai03

Reply:
Quick starts shouldn't be loaded like this directly, it skips the permissions check which is eseential to determine what quikc starts are available for the current user based on his permissions. Also, the permission checker cannot be a simple hook since the quick starts are coming as `ConsoleQuickStart` CRs and we have to run permission check for each of them use `useAccessReview` hook. We cannot run the access review check safely over an array of items that are coming from backend and can change anytime.